hack/update-staging-godeps-dockerized.sh

```diff
@@ -0,0 +1,116 @@
+KUBE_VERBOSE="${KUBE_VERBOSE:-5}"
+if (( ${KUBE_VERBOSE} >= 6 )); then
+  GODEP_OPTS+=("-v")
+fi
+
+while getopts ":df" opt; do
+  case $opt in
+    d) # do not godep-restore into a temporary directory, but use the existing GOPATH
+      DRY_RUN=true
+      ;;
+    f) # fail if something in the Godeps.json files changed
+      FAIL_ON_DIFF=true
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      exit 1
+      ;;
+  esac
+done
+
+kube::util::ensure_godep_version v79
+
+echo "Checking whether godeps are restored"
+if ! kube::util::godep_restored 2>&1 | sed 's/^/  /'; then
+  hack/godep-restore.sh
+fi
+
+kube::util::ensure-temp-dir
+TMP_GOPATH="${KUBE_TEMP}/go"
+
+function updateGodepManifest() {
+  local repo="${1}"
+  pushd "${TMP_GOPATH}/src/k8s.io/${repo}" >/dev/null
+    kube::log::status "Updating godeps for k8s.io/${repo}"
+    rm -rf Godeps # remove the current Godeps.json so we always rebuild it
+    GOPATH="${TMP_GOPATH}:${GOPATH}:${KUBE_ROOT}/staging" godep save ${GODEP_OPTS} ./... 2>&1 | sed 's/^/  /'
+
+    # Rewriting Godeps.json to remove commits that don't really exist because we haven't pushed the prereqs yet
+    go run "${KUBE_ROOT}/staging/godeps-json-updater.go" --godeps-file="${TMP_GOPATH}/src/k8s.io/${repo}/Godeps/Godeps.json" --override-import-path="k8s.io/${repo}"
+
+    # commit so that following repos do not see this repo as dirty
+    git add vendor >/dev/null
+    git commit -a -m "Updated Godeps.json" >/dev/null
+  popd >/dev/null
+}
+
+# move into staging and save the dependencies for everything in order
+mkdir -p "${TMP_GOPATH}/src/k8s.io"
+for repo in $(ls ${KUBE_ROOT}/staging/src/k8s.io); do
+  # we have to skip client-go because it does unusual manipulation of its godeps
+  if [ "${repo}" == "client-go" ]; then
+    continue
+  fi
+  # we skip metrics because it's synced to the real repo manually
+  if [ "${repo}" == "metrics" ]; then
+    continue
+  fi
+
+  kube::util::ensure_clean_working_dir
+
+  cp -a "${KUBE_ROOT}/staging/src/k8s.io/${repo}" "${TMP_GOPATH}/src/k8s.io/"
+
+  pushd "${TMP_GOPATH}/src/k8s.io/${repo}" >/dev/null
+    git init >/dev/null
+    git config --local user.email "nobody@k8s.io"
+    git config --local user.name "$0"
+    git add . >/dev/null
+    git commit -q -m "Snapshot" >/dev/null
+  popd >/dev/null
+
+  updateGodepManifest "${repo}"
+
+  if [ "${FAIL_ON_DIFF}" == true ]; then
+    diff --ignore-matching-lines='^\s*\"GoVersion\":' --ignore-matching-line='^\s*\"GodepVersion\":' --ignore-matching-lines='^\s*\"Comment\"' -u "${KUBE_ROOT}/staging/src/k8s.io/${repo}/Godeps" "${TMP_GOPATH}/src/k8s.io/${repo}/Godeps/Godeps.json"
+  fi
+  if [ "${DRY_RUN}" != true ]; then
+    cp "${TMP_GOPATH}/src/k8s.io/${repo}/Godeps/Godeps.json" "${KUBE_ROOT}/staging/src/k8s.io/${repo}/Godeps"
+    # Assume Godeps.json is not updated, so that it doesn't make the working tree dirty
+    git update-index --assume-unchanged "${KUBE_ROOT}/staging/src/k8s.io/${repo}/Godeps/Godeps.json"
```

Comment:
The godeps script requires a clean git tree. Right now, it requires you commit each file, and pauses on each repo in the run that has changes (it's a real PITA). What this does is ignores changes to the file we know is changing, but the git tree inside the container doesn't matter, as we rsync the changed files outs.